lib/ex_crypto.ex

```diff
@@ -36,6 +36,33 @@ defmodule ExCrypto do
     {kind, Exception.normalize(kind, error), System.stacktrace}
   end
 
+  @aes_block_size 16
+  @iv_bit_length 128
+
+  defp key_and_iv_for_algorithm(raw_key, raw_iv, algorithm) do
+    bits = algorithm_bits(algorithm)
+    {iv, _} = bitstring_size(raw_iv, @iv_bit_length)
+    {key, _} = bitstring_size(raw_key, bits)
+    {key, iv}
+  end
+
+  defp test_key_and_iv_against_algorithm(key, iv, algorithm) do
+    iv_size = bit_size(iv)
+    key_size = bit_size(key)
+    if iv_size != @iv_bit_length || key_size != algorithm_bits(algorithm) do
+      raise "IV must be exactly 128 bits and key must be exactly #{algorithm_bits(algorithm)} bits, or use encrypt!/decrypt! method"
+    end
+  end
+
+  defp algorithm_bits(algorithm) do
```

Comment:
yeah that's a good idea, simplicity wins over automation, and it's not like we need to support a dozen or more different keylengths, probably just 3 or 4 ever :-)  I often overcomplicate like this, agree the code was smelly, probably more of a fun exercise to write than a good idea